### PR TITLE
Improve geometry documentation on congruence, fixes #2984

### LINKED ISSUE #2984: tgsss1, tgsss2, and tgsss3 are not SSS as intended
Theorems `tgsss1`, `tgsss2`, and `tgsss3` are true, but I don't think they prove what was intended.

The comment currently describing theorem `tgsss1` says it is "Third congruence theorem:  SSS (Side-Side-Side):  If the three pairs of sides of two triangles are equal in length, then the triangles are congruent.  Theorem 11.51 of [Schwabhauser] p. 109.". However, `tgsss1` does *NOT* prove that given congruent segments you end up with ABC is-congruent-as-a-triangle with DEF. I would expect SSS to show that 2 triangles are congruent, and `tgsss1` doesn't do that. It instead proves that ABC is-congruent-as-an-angle with DEF. That is, `tgsss1` proves `<" A B C "> ( cgrA ` G ) <" D E F ">`, but I think it's supposed to instead prove `<" A B C "> ( cgrG ` G ) <" D E F ">` (note the use of `cgrG` instead of `cgrA`).

By using the Google lens app on Schwabhauser, I've become convinced that Schwabhauser also means to use `cgrG` and not `cgrA` for theorem 11.51.

I think we should keep these Metamath theorems (renaming them & changing their comment to match what they prove), but I think we separately need the intended SSS theorems from 11.51 that prove that certain *triangles* are congruent.

Am I misunderstanding something? @tirix @jkingdon @digama0 ?

### PR DESCRIPTION
**Pull request by david-a-wheeler**
Improve documentation about our approach to congruence in the geometry section.

* We encode congruence using a "distance" function. Make it clear that this encoding is justified by a formal proof, per tgjustc1 and tgjustc2 , which were added via <https://github.com/metamath/set.mm/pull/3014>
* We encode shape congruence as (`cgrG`) as all corresponding segments between all corresponding points are congruent. Make that clear, explain why that's done including its justification in `dfcgrg2` This builds on <https://github.com/metamath/set.mm/pull/2994>.

Signed-off-by: David A. Wheeler <dwheeler@dwheeler.com>

### CODE REVIEW

File: set.mm
Comment (benjub) [inline set.mm:381014]: Encoding congruence as distance equality makes...
or
Encoding congruence as an equality of distances makes...

File: set.mm
Comment (benjub) [inline set.mm:382528]: the relation OF congruence

File: set.mm
Comment (benjub) [inline set.mm:382530]: I saw in nearby comments the terms "shape" and "sequence". Say somewhere "shapes are finite sequences of points", or maybe simply remove the word "shape" everywhere and only use the term "finite sequence of points" (rather than "word"). @tirix will see what is best.

Comment (benjub) [reply set.mm:382530]: Also: maybe write "...if the segments formed on all pairs of corresponding points..." ?

File: set.mm
Comment (benjub) [inline set.mm:382540]: I think this is not the place to talk about triangles and even less angles. This definition is about the congruence of general shapes. The two primitive notions in Tarski geometry are betweenness and congruence of segments. Congruence of angles is only a derived notion defined later in terms of it, so it makes no sense to define angle congruence using angles.

File: set.mm
Comment (benjub) [inline set.mm:382531]: It looks like the current definition sees shapes as partial functions on RR.

File: set.mm
Comment (icecream17) [reply set.mm:382528]: congruence relation?

Comment (benjub) [reply set.mm:382528]: Probably both are correct, but the former has the advantage of having "congruence between shapes" as a substring.

File: set.mm
Comment (david-a-wheeler) [reply set.mm:382540]: I disagree. Many people are going to expect "shape congruence" to include angle congruence. This definition does *not* include the expected angle congruence, so we need to explain why not. A short justification, with links to relevant proofs, provides that justification.

Comment (benjub) [reply set.mm:382540]: Are you sure you're not confusing with Hilbert geometry ?  Hilbert geometry contains angle congruence as a primitive notion, but Tarski geometry does not. These "many books" and "many people" you're talking about are probably developing geometry in other systems than the one used in set.mm. If that is indeed what you mean, then one could write something like:
> Other axiomatic systems of geometry, like Hilbert's system, also have a primitive notion of angle congruence, and therefore, congruence of triangles (which are simply "shapes" with three points) is sometimes defined in these systems using both (segment) congruence and angle congruence.  In Tarski's system, angle congruence is a derived notion, defined in df-.... and theorems ... prove that congruent triangles have congruent angles [and also all the SAS and CPTC stuff].

File: set.mm
Comment (avekens) [reply set.mm:382530]: The section header says "series of points"... I think the term "shape" is more intuitive, so we should keep it, but we should define it as "sequence of points" (see the comment for ~iscgrg), as @benjub proposed

File: set.mm
Comment (avekens) [reply set.mm:382540]: Maybe the additional part of the comment (modified as @benjub proposed) should be moved to the section header.

File: set.mm
Comment (david-a-wheeler) [reply set.mm:381014]: I agree. I slightly prefer the second one, so I'll use that. New version coming soon :-).

File: set.mm
Comment (david-a-wheeler) [reply set.mm:382528]: Sounds good. I'll use "the relation OF congruence between shapes".

File: set.mm
Comment (david-a-wheeler) [reply set.mm:382530]: Good point (hah!). I'll add this text here: **A "shape" is a finite sequence of points.**

File: set.mm
Comment (tirix) [reply set.mm:382531]: The reason for using functions on `RR` was that it could be used to define congruence of more complex shapes, like maybe segments and circles (provided they have the same parametrization. In fact until now, the definition has only been used on triangles represented by their 3 vertices.

Comment (benjub) [reply set.mm:382531]: So why not take any common domain ? Or in the other direction (probably to be preferred), why not really take finite sequences i.e. `( 0 ..^ n )` for some `n e. _om` ? That way, you have a theory independent from RR, which is in the spirit of Tarski's FOL axiomatization of geometry (I think models are powers of real closed fields, not necessarily RR ?).

Other question: can you point me to the theorem that says that if the triples (abc) and (def) are congruent and if Babc then Bdef ? That theorem is the crucial one, it even justifies the phrase "congruent" applied to shapes. And from that, you of course obtain angle congruence.

(The idea is that "congruence" of two things means that all primitive notions agree on these two things. And obviously, then all derived notions also agree.)

Comment (david-a-wheeler) [reply set.mm:382531]: Can you suggest a better wording?

File: set.mm
Comment (david-a-wheeler) [reply set.mm:382540]: @benjub - I like the idea of your text explaining that the system we're using doesn't have angle congruence as a primitive notion. I'll take your text & modify it to try to make it "flow" in this discussion about definitions.

Even with that statement, though, we need to justify why we (and Tarski) are willing to define congruence of shapes without including congruence of angles. Many typical geometry materials, including those used worldwide in high school, include angles as a primitive fo defining the congruence of triangles. Oh, and don't say high school geometry is irrelevant; Tarski often taught high school geometry In the 1920s and 30s, and this led to his interest in geometry that we're formalizing. This looks like a big deviation, and we can better than an appeal to authority.

As evidence, here are some of sample materials I've found that *expressly* define congruence of triangles as requiring congruence of BOTH their corresponding angles AND their corresponding segments:

* "Two triangles are congruent if a one-to-one correspondence can be set up between their vertices so that corresponding sides are congruent and corresponding angles are congruent." - Euclidean and Non-Euclidean Geometries: Development and History by Marvin Jay Greenberg, 1974, page 69
* "The (example) two triangles are congruent because they are geometrically equivalent. All corresponding sides have equal lengths and all corresponding angles have equal measures." - Advanced Mathematics (Second Edition) by Saxon, Page 67.
* "Two triangles are congruent if their corresponding sides are equal in length, and their corresponding angles are equal in measure." - Wikipedia, [Congruence(geometry)](https://en.wikipedia.org/wiki/Congruence_(geometry)#:~:text=Congruence%20of%20triangles,-See%20also%3A%20Solution&text=Two%20triangles%20are%20congruent%20if,angles%20are%20equal%20in%20measure.)

It's *okay* that we don't include angles in the definition, there are even good reasons for it. We just need a brief comment explaining why why don't. Saying "that's what Tarski did" isn't enough, there's a good reason why Tarski did this that isn't merely an appeal to authority.

A brief note in the header as *well* would make sense, but people who surf this material though hyperlinks may end up on this definition of shape congruence without any context. Let's give them that context, and frankly, the details about shape congruence belong here, in the definition shape congruence.

File: set.mm
Comment (benjub) [reply set.mm:382531]: @david-a-wheeler : if the question is to me: no, I cannot ;)  I think the decision is in @tirix's hands and then he/we'll adapt the text to the definition he chooses.
In any case, I do not know why "ideally" is used: if it were ideal, then it is very easy to make the change. On the other hand, I have not practiced Tarski's geometry enough to know what is gained in considering infinite (and even non-discrete) shapes. Considering infinite shapes removes the feel of simplicity of Tarski's geometry.

**Comment by david-a-wheeler** [comment]
I've posted an update that I *think* addresses all comments here but one. I think it's important to clarify that this definition of shape congruence is different than some others', and provide the rationale.

The exception: "It looks like the current definition sees shapes as partial functions on RR." - do you have any suggested wording?

File: set.mm
Comment (benjub) [reply set.mm]: Thanks for the specific citations in these two books. I'll try to have a look because for now I do not know their axiom systems nor how they define "congruent angles".  By the way, I note that Greenberg indeed uses unordered shapes (i.e. finite sets as opposed to finite sequences), as I proposed in https://github.com/metamath/set.mm/pull/2994#discussion_r1083284350 before I saw that Schwabhauser uses sequences/tuples. I think the unordered shapes are more natural (a triangle does not care about the order in which you enumerate its vertices). But maybe that's too much worl for @tirix to modify it for a too little gain.

The "good reason why Tarski did it" is that angle congruence can be defined in terms of congruence and betweenness alone. So these two primitive notions suffice. This is connected to my question to @tirix in https://github.com/metamath/set.mm/pull/3028#discussion_r1096788692

As for the specific text, I'll let you @david-a-wheeler see, because it looks like you now have all the parts.

Comment (david-a-wheeler) [reply set.mm]: I think the ordered sequence representation (words) chosen by @tirix is a good one. If triangle ABC is congruent to DEF, that does *NOT* mean that triangle BAC is necessarily congruent. So while in some sense the "order doesn't matter", when *referring* to a particular shape for making congruence statements, you typically *do* identify the shape by listing a specific order. I view this as a "shorthand for clearly identifying a specific shape".

File: set.mm
Comment (david-a-wheeler) [reply set.mm]: The "Ideally..." text was in the original, I didn't write that (though it makes sense to me). If we don't have any specific changes to that sentence we want to make, I propose we just keep it as-is until we do :-).

Comment (tirix) [reply set.mm]: Let me paste Schwabäuser's definition below:
image
Which I would translate to:
> To describe more conveniently several congruences, we introduce the congruence of "geometrical figures" (made of finitely many points)
> 
> Definition: The n-tuples of points $< a_1,...,a_n >$ and $< b_1,...,b_n >$ are said to be congruent when: ... 

Word like `<" A B C ">` are the convenient way in Metamath to express tuples, so if we prefer to restrict ourselves strictly to Schwabhäuser's definition, the domain to be used instead of `RR` would be `NN0`, to match with the domain of words (see [df-word](https://us.metamath.org/mpeuni/df-word.html)). It would be much more convenient that `_om`.

**Comment by tirix** [comment]
> Other question: can you point me to the theorem that says that if the triples (abc) and (def) are congruent and if Babc then Bdef ? That theorem is the crucial one, it even justifies the phrase "congruent" applied to shapes. And from that, you of course obtain angle congruence.

That would be [tgbtwnxfr](https://us.metamath.org/mpeuni/tgbtwnxfr.html), _Theorem 4.6 of [[Schwabhauser](https://us.metamath.org/mpeuni/mmset.html#Schwabhauser)] p. 36._

File: set.mm
Comment (david-a-wheeler) [reply set.mm]: @tirix has responded why, and I think it's sensible.

@benjub : You said, "I think the decision is in @tirix's hands and then he/we'll adapt the text to the definition he chooses." Since this is the definition he chose, and has reaffirmed it here, I *think* this comment is resolved, so I'm resolving it. If I'm wrong please let us know :-).

File: set.mm
Comment (benjub) [reply set.mm]: That's why the definition that Greenberg and I (in https://github.com/metamath/set.mm/pull/2994#discussion_r1083284350) proposed are formulated using a bijection. A triangle is defined by a set of three points, not a sequence of three points.

File: set.mm
Comment (benjub) [reply set.mm]: Yes, I meant NN0 sorry. So the definition should be
```
... { <. a , b >. |
E. n e. NN0 ( ( a : ( 0 ..^ n ) --> ( Base ` g ) /\ b : ( 0 ..^ n ) ->  ( Base ` g ) ) /\ ...
```

**Comment by benjub** [comment]
> > Other question: can you point me to the theorem that says that if the triples (abc) and (def) are congruent and if Babc then Bdef ? That theorem is the crucial one, it even justifies the phrase "congruent" applied to shapes. And from that, you of course obtain angle congruence.
> 
> That would be [tgbtwnxfr](https://us.metamath.org/mpeuni/tgbtwnxfr.html), _Theorem 4.6 of [[Schwabhauser](https://us.metamath.org/mpeuni/mmset.html#Schwabhauser)] p. 36._

Thanks @tirix. That theorem, via its corollary:

If $(a_0, \dots, a_{n-1}) \equiv (b_0, \dots, b_{n-1})$ and $0 \leq i, j, k < n$, then $B a_i a_j a_k$ iff $B b_i b_j b_k$.

is fundamental. If it were not true, then it should be added to the definition of congruence of shapes. I don't know how you emphasized it in comments, but it is really important.

(The same thing could be said of equality: If $(a_0, \dots, a_{n-1}) \equiv (b_0, \dots, b_{n-1})$ and $0 \leq i, j < n$, then $a_i = a_j$ iff $b_i = b_j$. This is true by one of the first axioms of Tarski geometry, but again if there was no such axiom, then that property should be added to the definition of congruence.)

File: set.mm
Comment (david-a-wheeler) [reply set.mm]: > I'll try to have a look because for now I do not know their axiom systems nor how they define "congruent angles". 

The *usual* definition is, I believe, "congruent angles have equal measures".

In any case, I hope I've convinced you that there are *other* systems that regularly refer to "congruent angles" when they define "congruent shapes/triangles". All I want to do is provide a brief justification for the decision to *not* do that here, along with links to relevant proofs. It's not a change in the definition, but an explanation to help the reader understand why this specific definition is used.

**Comment by david-a-wheeler** [comment]
I *think* I've addressed all comments/issues. If not, or if there are new issues, please raise them!

**Review by benjub — APPROVED** [review]
Good for me, though it might be wise to wait for @tirix's approval.

File: set.mm
Comment (tirix) [inline set.mm:382556]: Actually, words are defined as functions over `NN0`, not over `NN`.

File: set.mm
Comment (benjub) [reply set.mm]: Then what is the "measure" of an angle ! This is actually trickier than one may think. (Actually, the "measure" of an angle is often defined to be its equivalence class for the angle congruence relation, so it's the other way around; it's similar to a "distance" being defined as an equivalence class for the segment congruence relation.)

I know there are other systems that have angle congruence as a primitive notion (I mentioned Hilbert's as such a system above); my point is that they are *other* systems. I agree with the explanation that you now wrote (my only problem with the first version was that the distinction between the different systems was not clear).

File: set.mm
Comment (benjub) [reply set.mm:382556]: well... functions over `( 0 ..^ n )` for some n \in NN0 !

Comment (tirix) [reply set.mm:382556]: Indeed! Maybe just refer to ~ df-word ?
Something like 
> but we will use words (see ~ df-word) in most cases.

Comment (david-a-wheeler) [reply set.mm:382556]: I like that, it's both clear and accurate (which is often hard to do). I'll post an update with that.

File: set.mm
Comment (david-a-wheeler) [reply set.mm]: @benjub :
> I agree with the explanation that you now wrote (my only problem with the first version was that the distinction between the different systems was not clear).

Excellent! Good to hear, I think the results are better than my first try for sure. Thank you! Given what you just said, I'm marking this as "resolved".

**Comment by david-a-wheeler** [comment]
I think we're done!! Any more comments? Otherwise I think it's ready to merge.

**Review by avekens — APPROVED** [review]

File: set.mm
Comment (avekens) [reply set.mm]: > Let me paste Schwabäuser's definition below: image Which I would translate to:
> 
> > To describe more conveniently several congruences, we introduce the congruence of "geometrical figures" (made of finitely many points)
> > Definition: The n-tuples of points <a1,...,an> and <b1,...,bn> are said to be congruent when: ...
> 

The German "geometrische Figuren" could also be translated to "geometric shapes", which fits to our chosen terminology.
Furthermore, "Streckenkongruenz" does not only mean "congruence ", but "congruence of distances" - also perfectly adopted in our concepts/comments. So everything is fine now!

**Review by tirix — APPROVED** [review]
Looks good to me! Thank you David!

**Comment by david-a-wheeler** [comment]
Great!  And @tirix - thanks to *you*. You did a lot of amazing things, I'm just trying to help people better understand all this great work.

**Comment by david-a-wheeler** [comment]
@tirix @benjub - thank you both!!

**Pull request merged**
